### PR TITLE
BE-8107 redis cluster

### PR DESCRIPTION
**Pull request by waitstime**
Update to pass in a Redis interface to use for the Redis checkpoint rather than passing a URL used to create a redis client directly. Still maintains default behavior to create a local Redis client if nothing specified but instead use nrredis.

Change log package to use gametimesf/log

### CODE REVIEW

**Comment by ajanata** [comment]
This is a fork of a public repo; if we make these changes we'll never be able to catch up to upstream again. Do we really need to make these changes, and if so, should we catch up to upstream first?

File: config.go [hunk 1]
```diff
@@ -114,13 +107,7 @@ func (c *Config) setDefaults() {
...
-		Addr: redisURL,
-	})
-	_, err := client.Ping().Result()
-	if err != nil {
-		return nil, err
-	}
-	return client, nil
+func redisClientDefault() *nrredis.ClusterClient {
+	options := nrredis.NewClusterModeDisabledOptions([]string{defaultRedisAddr})
+	return nrredis.NewClusterClient(options)
```
Comment (nmwari) [inline config.go:112]: This looks like a little change in behavior. It looks live previously we were making sure the connection was successful by pinging. Should we keep the old behavior? Or is there a good reason to remove that behavior?

File: redis_checkpoint_test.go [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
-	client := redis.NewClient(&redis.Options{Addr: defaultAddr})
+	options := nrredis.NewClusterModeDisabledOptions([]string{defaultAddr})
+	client := nrredis.NewClusterClient(options)
 
 	c := RedisCheckpoint{
 		AppName:    "app",
 		StreamName: "stream",
 		client:     client,
 	}
+	c.sequenceNumbers = make(map[string]string)
```
Comment (nmwari) [inline redis_checkpoint_test.go:21]: Why is this necessary now?

**Review by nmwari — COMMENTED** [review]
Log changes seem good to me, just want to make sure I'm understanding some of the redis changes

File: redis_checkpoint_test.go (hunk 2 above)
Comment (waitstime) [reply redis_checkpoint_test.go:21]: It was panicking before. The function where this was created doesn't get called with the way we're manually setting this up.

**Comment by waitstime** [comment]
Closing this and created https://gametime.atlassian.net/browse/BE-8380. Discussed in Platform stand that pulling in latest upstream changes will likely help us avoid needing to do these updates and uses go-redis/v8 which supports connecting to redis clusters in cluster mode disabled.
